debug_module.tex

```diff
@@ -166,25 +166,29 @@ \section{Hart States}
 \section{Run Control} \label{runcontrol}
 
 For every hart, the Debug Module contains 4 conceptual bits of state: halt
-request, resume request, halt-on-reset request,  and hart reset.
-(The hart reset and halt-on-reset request bits are optional.)
-These bits all reset to 0. A debugger can write them for the currently selected
-harts through \Fhaltreq, \Fresumereq, \Fsetresethaltreq/\Fclrresethaltreq and
-\Fhartreset in \Rdmcontrol, only one of which may be high in any single write.
-In addition the DM receives halted, running, and resume ack signals from each
-hart.
-
-When a running hart receives a halt request, it responds by halting,
-deasserting its running signal, and asserting its halted signal. The halted
-signals of all selected harts are reflected in the \Fallhalted and \Fanyhalted
-bits. \Fhaltreq is ignored by halted harts.
-
-When a halted hart receives a resume request, it responds by resuming, clearing
-its halted signal, and asserting its running signal and resume ack signals. The
-resume ack signal is lowered when the resume request is deasserted.  These
```

Comment:
this is the only thing in the pre-PR version that I object to. I would just say resume ack is deasserted as soon as another resumereq is set. Why are we rewriting this entire section?

Reply:
The new version includes what actions the debugger takes to get here, separating out a bit more clearly what is going on between the debugger, the DM, and the hart.

It also makes more explicit that the DM tracks resumeack, not the hart. This allows us to say that the DM clears resumeack when resumereq is written 1. Otherwise it implies some round-trip to the hart, which has the race mentioned earlier.